boards/common/esp8266/board_common.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    boards_common_esp8266 ESP8266 Common
+ * @ingroup     boards_common
+ * @brief       Common files for the esp8266 board.
+ * @{
+ *
+ * @file
+ * @brief       Definitions for all esp8266 board.
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include "espressif/eagle_soc.h"
+#include "board_common.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+static uint32_t leds_initialized = 0x0;
+
+void led_on_off (uint8_t led, uint8_t value)
+{
+    uint32_t led_mask = BIT(led);
+    if (!(leds_initialized & led_mask)) {
+        gpio_init (led, GPIO_OUT);
+        leds_initialized |= led_mask;
+    }
+    GPIO_REG_WRITE((value ? GPIO_OUT_W1TC_ADDRESS : GPIO_OUT_W1TS_ADDRESS), led_mask);
+}
+
+void led_toggle (uint8_t led)
+{
+    uint32_t led_mask = BIT(led);
+    led_on_off (led, ((GPIO_REG_READ(GPIO_OUT_ADDRESS) & led_mask)) ? 1 : 0);
+}
+
+void board_init(void)
+{
+    #if defined(SPI_USED)
```

Comment:
this should be covered by periph_common init.c

Reply:
@kaspar030 When I went through existing CPU / board directories, it did not become really clear to me which parts of the source code should be placed in which file The Porting Guide on the wiki does not really help: "There is however no generic way to structure the code, as CPU architectures and families can be very different."

According to the API reference ```spi_init``` should be called by a board during system initialization. So I though function ```board_init``` is the right place. I have not seen a file ```periph_common_init.c``` anywhere.

In difference to other architectures, available peripherals are not determined by the board but by the software. The MCU just have a number of GPIOs for different purposes that are all simply broken out by all the boards I know, except the ESP-01. That is, normally board doesn't decide whether the GPIOs are used for SPI but the software.

Reply:
@kaspar030 Ok, I got it. I will remove it from my ```board_init```.